src/app/perspective-status.component.ts

```diff
@@ -197,6 +195,113 @@ export class PerspectiveStatus implements OnChanges {
+      this.updateStatusWidgetVisibilityAnimation.kill();
+    }
+
+    // If nothing has changed, return an empty animation.
+    if (hide === this.shouldHideStatusWidget) {
+      console.debug('Returning without update status widget visibility animation');
+      return new TimelineMax({});
+    } else {
+      console.debug('Getting update status widget visibility animation');
+    }
+
+    this.updateStatusWidgetVisibilityAnimation = new TimelineMax({
+      onStart: () => {
+        this.ngZone.run(() => {
+          this.isPlayingShowOrHideLoadingWidgetAnimation = true;
+          console.debug('Updating status widget visibility to '
+                        + (hide ? 'hidden' : 'visible') + ' from '
+                        + (this.shouldHideStatusWidget ? 'hidden' : 'visible'));
+          // Disable hiding so animations will show up.
+          this.shouldHideStatusWidget = false;
+        });
+      },
+      onComplete: () => {
+        this.ngZone.run(() => {
+          console.debug('Changing status widget visibility complete');
+          this.isPlayingShowOrHideLoadingWidgetAnimation = false;
+          this.shouldHideStatusWidget = hide;
+        });
+      },
+    });
+    this.updateStatusWidgetVisibilityAnimation.add([
+      this.getChangeLoadingIconVisibilityAnimation(hide),
+      this.getChangeLoadingIconXValueAnimation(hide)]);
+    return this.updateStatusWidgetVisibilityAnimation;
+  }
+
+  private getChangeLoadingIconVisibilityAnimation(hide: boolean): TweenMax {
+    return TweenMax.to(
+      this.widget, FADE_WIDGET_TIME_SECONDS, { opacity: hide ? 0 : 1})
+  }
+
+  private getChangeLoadingIconXValueAnimation(hide: boolean): TimelineMax {
+    let timeline = new TimelineMax({});
+    let translateXAnimations: Animation[] = [];
+    translateXAnimations.push(
+      TweenMax.to(this.widget, FADE_WIDGET_TIME_SECONDS,
+                  { x: hide ? -1 * (this.indicatorWidth
+                                    + WIDGET_PADDING_PX
+                                    + WIDGET_RIGHT_MARGIN_PX)
+                            : 0}));
+    if (this.configuration === Configuration.DEMO_SITE) {
+      // Also shift the text for the leftmost element in each layer left/right
+      // as needed. Even though only layer 0 is visible when the score changes,
+      // the elements in the rest of the layers need to be adjusted to match
+      // for when we transition to other layers.
+      let layer0TextContainer = this.elementRef.nativeElement.querySelector(
+          this.layerAnimationSelectors[0] + ' .layerText');
+      let layer1TextContainer = this.elementRef.nativeElement.querySelector(
+          this.layerAnimationSelectors[1] + ' .layerText');
+      let layer2InteractiveContainer =
+        this.elementRef.nativeElement.querySelector(
+          this.layerAnimationSelectors[2] + ' .interactiveElement');
+      translateXAnimations.push(
+        TweenMax.to(layer0TextContainer, FADE_WIDGET_TIME_SECONDS,
```

Comment:
It looks like these are all the same except for which element they are being applied to, would it make sense to either set the animation property as a constant or have a function to generate it?

Reply:
Moved into a const